ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py

```diff
@@ -229,69 +220,62 @@ def check_for_halted_volumes(logger):
         :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
         """
 
-        logger.info("Checking for halted volumes: ")
+        logger.info('Checking for halted volumes: ')
 
         vpools = VPoolHelper.get_vpools()
 
         if len(vpools) != 0:
 
             for vp in vpools:
 
-                if vp.guid in VolumedriverHealthCheck.MACHINE_DETAILS.vpools_guids:
-
-                    haltedvolumes = []
+                if vp.guid not in VolumedriverHealthCheck.LOCAL_SR.vpools_guids:
+                    logger.skip('Skipping vPool {0} because it is not living here.'.format(vp.name), 'halted_volumedriver_{0}'.format(vp.name))
+                    continue
 
-                    logger.info("Checking vPool '{0}': ".format(vp.name))
+                haltedvolumes = []
+                logger.info('Checking vPool {0}: '.format(vp.name))
+                config_file = Configuration.get_configuration_path("/ovs/vpools/{0}/hosts/{1}{2}/config".format(vp.guid, vp.name, VolumedriverHealthCheck.LOCAL_ID))
```

Comment:
Also here don't create the storagedriver name yourself, because if we ever change this, healthcheck won't work.
Use the storagedriver.name property